x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/GeoLineAggregatorFactory.java

```diff
@@ -58,7 +58,11 @@ protected Aggregator doCreateInternal(
         Map<String, Object> metaData
     ) throws IOException {
         GeoLineMultiValuesSource valuesSources = new GeoLineMultiValuesSource(configs);
-        return new GeoLineAggregator(name, valuesSources, context, parent, metaData, includeSort, sortOrder, size);
+        if (context.isInSortOrderExecutionRequired()) {
```

Comment:
Are we checking somewhere else that the provided sort is @timestamp? if it is not then we should not use this path?

Reply:
You are right, I do not yet do the sort-field check. I had a comment in the description about missing negative tests, but for sure this is also missing code, so definitely needs to be added.

Reply:
OK. Now this is done. I check that the sort field is `@timestamp` the hard-coded field for time-series data stream, and only if that is true allow the geo_line optimization. I also enhanced the tests to run with the field as either `@timestamp` or `time_field` and verify different behaviour for the two cases.

For `position` metrics, we can disallow the sort-field entirely. That can be a followup PR.